apps/examples/testcase/le_tc/kernel/tc_irq.c

```diff
@@ -0,0 +1,81 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file tc_irq.c
+
+/// @brief Test Case Example for Errno API
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <tinyara/timer.h>
+#include <tinyara/irq.h>
+#include "tc_internal.h"
+
+#  define TIMER_DEVNAME  "/dev/timer%d"
+
+/**
+ * @fn                  :tc_irq
+ * @brief               :test the fin_wait using timer
+ * @scenario            :none
+ * API's covered        :fin_wait
+ * Preconditions        :none
+ * Postconditions       :none
+ * @return              :void
+ */
+
+#ifdef CONFIG_TIMER
```

Comment:
We should make a dependency with timer in Kconfig, not here. Because even user enables irq tc, because of missing timer enabled, nothing is executed. It makes user confused.
And when we enable irq tc config and disable timer config, this makes compilation error.

Reply:
thanks for pointing it out. corrected now. please check now.

Reply:
@sunghan-chang i have added CONFIG_TIMER again because if we enable the CONFIG_TIMER then CONFIG_TC_KERNEL_IRQ would be enabled and written in .config but when we disable CONFIG_TIMER then CONFIG_TC_KERNEL_IRQ is still in the .config then it gives the compilation errors.
let me know your views.

Reply:
@deepaksrma It's impossible to disable CONFIG_TIMER even CONFIG_TC_KERNEL_IRQ is enabled. When you disable CONFIG_TIMER, then CONFIG_TC_KERNEL_IRQ will be disable automatically. That's why we make a dependency with TIMER in Kconfig. I think you consider manual modification for .config. But it should not be. 
```
#
# Automatically generated file; DO NOT EDIT.
# TinyAra Configuration
#
```
This is the first sentence of defconfig and .config. We should use menuconfig to modify configs.

Reply:
`When you disable CONFIG_TIMER, then CONFIG_TC_KERNEL_IRQ will be disabled automatically.`
i was also expecting the same but this is not happening and that is my point.

once you enable the CONFIG_TIMER then CONFIG_TC_KERNEL_IRQ is also enabled and now suppose if i disable CONFIG_TIMER still i see `CONFIG_TC_KERNEL_IRQ=y` in .config, which should not have happened, then it is resulting into compilation errors.

P.S. I am not modifying .config manually, i am modifying with menuconfig only.

Reply:
I'm not sure why you said like above.
When I test, if I disable the `CONFIG_TIMER`, then `CONFIG_TC_KERNEL_IRQ` is also disabled properly.
Please check again.

Reply:
CONFIG_TC_KERNEL_IRQ was not disabled properly because
i wrote `select TC_KERNEL_IRQ` 
instead of
`select TC_KERNEL_IRQ if TIMER`

now corrected and working fine.